include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,197 @@
+#ifndef XSPARSE_CO_ITERATION_HPP
+#define XSPARSE_CO_ITERATION_HPP
+
+#include <vector>
+#include <tuple>
+#include <algorithm>
+#include <climits>
+
+namespace xsparse::level_capabilities
+{
+    template <class F, class IK, class PK, class Levels, class Is>
+    class Coiterate;
+
+    template <class F, class IK, class PK, class... Levels, class... Is>
+    class Coiterate<F, IK, PK, std::tuple<Levels...>, std::tuple<Is...>>
+    {
+    private:
+        std::tuple<Levels&...> const m_levelsTuple;
+
+    public:
+        explicit inline Coiterate(Levels&... levels) noexcept
+            : m_levelsTuple(std::tie(levels...))
+        {
+        }
+
+    public:
+        class coiteration_helper
+        {
+        public:
+            Coiterate const& m_coiterate;
+            F const m_comparisonHelper;
```

Comment:
Please move this to `CoIterate`

Reply:
Should I move the `comparisonHelper` to Coiterate?

Reply:
Yes.